modules/table/src/test/java/org/apache/ignite/internal/table/distributed/TableUtilsTest.java

```diff
@@ -35,95 +37,163 @@
-        });
+    private final CatalogManager catalogManager = CatalogTestUtils.createTestCatalogManager("test-node", clock);
+
+    @BeforeEach
+    void setUp() {
+        assertThat(catalogManager.start(), willCompleteSuccessfully());
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        closeAll(
+                catalogManager::beforeNodeStop,
+                catalogManager::stop
+        );
     }
 
-    private void withCatalogManager(Consumer<CatalogManager> consumer) throws Exception {
-        CatalogManager catalogManager = CatalogTestUtils.createTestCatalogManager("test-node", clock);
+    @Test
+    void testIndexIdsAtRwTxBeginTs() {
+        createSimpleTable(catalogManager, TABLE_NAME);
 
-        assertThat(catalogManager.start(), willCompleteSuccessfully());
+        String indexName0 = INDEX_NAME + 0;
+        String indexName1 = INDEX_NAME + 1;
+        String indexName2 = INDEX_NAME + 2;
+        String indexName3 = INDEX_NAME + 3;
+        String indexName4 = INDEX_NAME + 4;
 
-        try {
-            consumer.accept(catalogManager);
-        } finally {
-            closeAll(catalogManager::beforeNodeStop, catalogManager::stop);
+        for (String indexName : List.of(indexName0, indexName1, indexName2, indexName3, indexName4)) {
+            createSimpleHashIndex(catalogManager, TABLE_NAME, indexName);
         }
+
+        int indexId0 = indexId(indexName0);
+        int indexId1 = indexId(indexName1);
+        int indexId2 = indexId(indexName2);
+        int indexId3 = indexId(indexName3);
+        int indexId4 = indexId(indexName4);
+
+        for (String indexName : List.of(indexName1, indexName2, indexName3, indexName4)) {
+            startBuildingIndex(catalogManager, indexId(indexName));
+        }
+
+        for (String indexName : List.of(indexName2, indexName3, indexName4)) {
+            makeIndexAvailable(catalogManager, indexId(indexName));
+        }
+
+        for (String indexName : List.of(indexName3, indexName4)) {
+            dropIndex(catalogManager, DEFAULT_SCHEMA_NAME, indexName);
+        }
+
+        removeIndex(catalogManager, indexId4);
+
+        CatalogManager spy = spy(catalogManager);
+
+        HybridTimestamp beginTs = clock.now();
+
+        int tableId = getTableIdStrict(catalogManager, TABLE_NAME, clock.nowLong());
+
+        assertThat(
+                indexIdsAtRwTxBeginTs(spy, transactionId(beginTs, 1), tableId),
+                contains(
+                        indexId(pkIndexName(TABLE_NAME)),
+                        indexId0,
+                        indexId1,
+                        indexId2,
+                        indexId3
+                )
+        );
+
+        verify(spy).activeCatalogVersion(eq(beginTs.longValue()));
+        verify(spy).indexes(eq(catalogManager.activeCatalogVersion(beginTs.longValue())), eq(tableId));
+    }
+
+    @Test
+    void testDroppedTables() {
```

Comment:
You should also try altering a table (so that the catalog contains multiple table descriptors for the same table ID) and check the result

Reply:
Ok